### DISCUSSION TITLE
custom APIRoute is not work as expected

### DISCUSSION THREAD

**Discussion by zffocussss** [fastapi/fastapi#9454] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.routing import APIRoute
from fastapi import Body, FastAPI, Request, Response
from typing import Callable, List

class GzipRoute(APIRoute):
    def get_route_handler(self) -> Callable:
        original_route_handler = super().get_route_handler()

        async def custom_route_handler(request: Request) -> Response:
            print("custom_route_handler called")
            return await original_route_handler(request)

        return custom_route_handler

app = FastAPI()
app.router.router_class = GzipRoute

@app.get("/")
async def root():
    return {"message": "Hello World"}
```

### Description

as [this link](https://fastapi.tiangolo.com/advanced/custom-request-and-route/) describe, user can get request body and do additional tasks before request go to the bussiness router functions.but it did not work in my case.I suppose it should call the function custom_route_handler,

### Operating System

macOS

### Operating System Details

macos M2

### FastAPI Version

0.86.0

### Python Version

3.11

**Comment by zffocussss** [comment]
the server side do not print my test string
![image](https://user-images.githubusercontent.com/46145695/234875442-296fb834-6489-4e63-93fb-33059bbec2e8.png)

**Comment by mateoradman** [comment, marked as answer]
Hi @zffocussss,

you have a typo in your code. 
`app.router.router_class = ...` should be `app.router.route_class = ...`

**Reply by zffocussss** [reply]
oh my god. it is not explicit to find the typo.

**Reply by zffocussss** [reply]
it works,thanks.
